enh(epp): operatingsystems-windows-snmp (#2482)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/operatingsystems-windows-snmp.md

|:-----------------|:------|
| swap.usage.bytes |       |

</TabItem>
<TabItem value="Traffic-*" label="Traffic-*">

| Metric name                                               | Unit  |
|:----------------------------------------------------------|:------|
| interface status                                          |       |
| *interface_name*#interface.traffic.in.bitspersecond       | b/s   |
| *interface_name*#interface.traffic.out.bitspersecond      | b/s   |
| *interface_name*#interface.packets.in.error.percentage    | %     |
| *interface_name*#interface.packets.in.discard.percentage  | %     |
| *interface_name*#interface.packets.out.error.percentage   | %     |
| *interface_name*#interface.packets.out.discard.percentage | %     |

> Applies to the following service templates: Traffic-Generic-Id, Traffic-Generic-Name, Traffic-Global

</TabItem>
<TabItem value="Uptime" label="Uptime">

| Metric name           | Unit  |
|:----------------------|:------|
| system.uptime.seconds | s     |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
</Tabs>

## Prerequisites

### SNMP Configuration

> To use this pack, the SNMP service must be properly configured on your ressource.
> Please refer to the official documentation from the constructor.

Reviewer: ```suggestion
To use this pack, the SNMP service must be properly configured on your Windows server. Please refer to the [Microsoft Knowledgebase to configure SNMP on your server](https://support.microsoft.com/en-us/kb/324263)
```

id: operatingsystems-windows-snmp
title: Windows SNMP
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Pack assets

### Templates

The Monitoring Connector **Windows SNMP** brings a host template:

* **OS-Windows-SNMP-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="OS-Windows-SNMP-custom" label="OS-Windows-SNMP-custom">

| Service Alias | Service Template              | Service Description                                                                                                                                  |
|:--------------|:------------------------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------|

> | Cpu           | OS-Windows-Cpu-SNMP-custom    | Check the rate of utilization of CPU for the machine. This check can give the average CPU utilization rate and the rate per CPU for multi-core CPU.  |

Reviewer 2: ```suggestion
| Cpu           | OS-Windows-Cpu-SNMP-custom    | Check the rate of utilization of the CPU for the machine. This check can give the average CPU utilization rate and the rate per CPU for multi-core CPUs.  |
```

| Memory        | OS-Windows-Memory-SNMP-custom | Check the rate of the utilization of memory                                                                                                          |
| Swap          | OS-Windows-Swap-SNMP-custom   | Check the rate of the utilization of virtual memory                                                                                                  |

> The services listed above are created automatically when the **OS-Windows-SNMP-custom** host template is used.

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias        | Service Template                            | Service Description                                                                                     | Discovery  |
|:---------------------|:--------------------------------------------|:--------------------------------------------------------------------------------------------------------|:----------:|

> | Disk-Generic-Id      | OS-Windows-Disk-Generic-Id-SNMP-custom      | Check the rate of free space on the disk. For each checks the name of the disk will appear              |            |

Reviewer 2: ```suggestion
| Disk-Generic-Id      | OS-Windows-Disk-Generic-Id-SNMP-custom      | Check the rate of free space on the disk. For each check the name of the disk will appear              |            |
```

Reviewer 2: ID?

> | Disk-Generic-Name    | OS-Windows-Disk-Generic-Name-SNMP-custom    | Check the rate of free space on the disk. For each checks the name of the disk will appear              |            |

Reviewer 2: ```suggestion
| Disk-Generic-Name    | OS-Windows-Disk-Generic-Name-SNMP-custom    | Check the rate of free space on the disk. For each check the name of the disk will appear              |            |
```

| Traffic-Global       | OS-Windows-Traffic-Global-SNMP-custom       | Check the bandwidth of the interface. For each checks the name of the interface will appear             | X          |
| Uptime               | OS-Windows-Uptime-SNMP-custom               | Check the uptime of the Windows server since the last reboot. It's just an indication with no threshold |            |

> The services listed above are not created automatically when a host template is applied. To use them, [create a service manually](/docs/monitoring/basic-objects/services), then apply the service template you want.

> If **Discovery** is checked, it means a service discovery rule exists for this service template.

</TabItem>
</Tabs>

### Discovery rules

#### Host discovery

| Rule name       | Description                                                                                                                                          |
|:----------------|:-----------------------------------------------------------------------------------------------------------------------------------------------------|

> | SNMP Agents     | Discover your resource through a SNMP subnet scan. You need to install the [Generic SNMP](./applications-protocol-snmp.md) to get the discovery rule |

Reviewer 2: ```suggestion
| SNMP Agents     | Discover your resources through a SNMP subnet scan. You need to install the [Generic SNMP](./applications-protocol-snmp.md) connector to get the discovery rule |
```

| WARNING      | Warning threshold average CPU utilization                                                           | 80                |             |
| CRITICAL     | Critical threshold average CPU utilization                                                          | 90                |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (E.g. a --verbose flag). All options are listed [here](#available-options) |                   |             |

</TabItem>
<TabItem value="Disk-Generic-Id" label="Disk-Generic-Id">

| Macro        | Description                                                                                         | Default value                                 | Mandatory   |
|:-------------|:----------------------------------------------------------------------------------------------------|:----------------------------------------------|:-----------:|
| TRANSFORMSRC | Regexp src to transform display value. (security risk!!!)                                           | ^(..).*                                       |             |

> | TRANSFORMDST | Regexp dst to transform display value. (security risk!!!)                                           | $1                                            |             |

Reviewer 2: explain more

> | DISKID       | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    |                                               |             |

Reviewer 2: ```suggestion
| DISKID       | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storages')                    |                                               |             |
```

| WARNING      | Warning threshold                                                                                   | 80                                            |             |
| CRITICAL     | Critical threshold                                                                                  | 90                                            |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (E.g. a --verbose flag). All options are listed [here](#available-options) | --filter-perfdata='storage.space\|used\|free' |             |

</TabItem>
<TabItem value="Disk-Generic-Name" label="Disk-Generic-Name">

| Macro        | Description                                                                                         | Default value                                 | Mandatory   |
|:-------------|:----------------------------------------------------------------------------------------------------|:----------------------------------------------|:-----------:|

> | TRANSFORMSRC | Regexp src to transform display value. (security risk!!!)                                           | ^(..).*                                       |             |

Reviewer 2: ?

> | TRANSFORMDST | Regexp dst to transform display value. (security risk!!!)                                           | $1                                            |             |

Reviewer 2: ?

> | DISKNAME     | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    |                                               |             |

Reviewer 2: ```suggestion
| DISKNAME     | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storages')                    |                                               |             |
```

| WARNING      | Warning threshold                                                                                   | 80                                            |             |
| CRITICAL     | Critical threshold                                                                                  | 90                                            |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (E.g. a --verbose flag). All options are listed [here](#available-options) | --filter-perfdata='storage.space\|used\|free' |             |

</TabItem>
<TabItem value="Disk-Global" label="Disk-Global">

| Macro        | Description                                                                                         | Default value                                           | Mandatory   |
|:-------------|:----------------------------------------------------------------------------------------------------|:--------------------------------------------------------|:-----------:|

> | FILTER       | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storage')                    | .*                                                      |             |

Reviewer 2: ```suggestion
| FILTER       | Set the storage (number expected) ex: 1, 2,... (empty means 'check all storages')                    | .*                                                      |             |
```

> | TRANSFORMSRC | Regexp src to transform display value. (security risk!!!)                                           | ^(..).*                                                 |             |

Reviewer 2: ?

> | TRANSFORMDST | Regexp dst to transform display value. (security risk!!!)                                           | $1                                                      |             |

Reviewer 2: ?

| Macro        | Description                                                                                         | Default value     | Mandatory   |
|:-------------|:----------------------------------------------------------------------------------------------------|:------------------|:-----------:|
| WARNING      | Warning threshold                                                                                   | 80                |             |
| CRITICAL     | Critical threshold                                                                                  | 90                |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (E.g. a --verbose flag). All options are listed [here](#available-options) |                   |             |

</TabItem>
<TabItem value="Ntp" label="Ntp">

| Macro        | Description                                                                                                     | Default value     | Mandatory   |
|:-------------|:----------------------------------------------------------------------------------------------------------------|:------------------|:-----------:|

> | NTPADDR      | Set the ntp hostname (if not set, localtime is used)                                                            |                   |             |

Reviewer 2: ```suggestion
| NTPADDR      | Set the NTP hostname (if not set, localtime is used)                                                            |                   |             |
```

> | NTPPORT      | Set the ntp port (Default: 123)                                                                                 |                   |             |

Reviewer 2: ```suggestion
| NTPPORT      | Set the NTP port (Default: 123)                                                                                 |                   |             |
```

> | TIMEZONE     | Set the timezone of distant server. For Windows, you need to set it. Can use format: 'Europe/London' or '+0100' |                   |             |

Reviewer 2: ```suggestion
| TIMEZONE     | Set the timezone of distant server. For Windows, you need to set it. You can use the following format: 'Europe/London' or '+0100' |                   |             |
```

#### Modes options

All available options for each service template are listed below:

<Tabs groupId="sync">
<TabItem value="Cpu" label="Cpu">

| Option             | Description                                   |
|:-------------------|:----------------------------------------------|

> | --use-ucd          | Use UCD mib for cpu average.                  |

Reviewer 2: ```suggestion
| --use-ucd          | Use UCD MIB for CPU average.                  |
```
